Install Musl for Rust Images

naftulikay opened the pull request:
musl is a tiny libc implementation and helps create cross-distro static binaries.

<!-- Thanks for contributing to _circleci-images_! Before you submit your pull request, please make sure to check the following boxes by putting an x in the [ ] -->

### Checklist

- [x] I've run `make` from the root directory to see all Dockerfiles are created successfully.
- [x] I've updated the documentation if necessary.

### Motivation and Context

A common use-case in Rust is to use [musl](https://www.musl-libc.org/) as a tiny `libc` implementation to build static binaries without any share library linkage, similar to the binaries that Golang produces by default. In order to compile these binaries, the following criteria should be met:

 1. `rustup target add x86_64-unknown-linux-musl`: add the toolchain
 2. install `musl` to add a `musl-gcc` binary for actually doing the compilation.

Since `circleci/rust` does not do any Rust-specific customizations to the base image, I have not added the toolchain but I have installed the musl packages to make static binary compilation possible.

### Description

This change simply installs the `musl`, `musl-dev`, and `musl-tools` into the image to make compilation to static `x86_64-unknown-linux-musl` binaries possible, provided that the Rust musl toolchain is installed for the given architecture.

iynere commented:
this is awesome, thanks so much for including test coverage in your PR ! @naftulikay

naftulikay commented:
praise be goss, tester of systems :pray:

iynere commented:
@naftulikay we get a lot of ephemeral failures in these jobs b/c of the sheer numbers of tags we build 😩 lemme rerun the failed jobs & see what's up. none of your rust tests actually failed tho, which is a good sign!

felicianotech commented:
Picking this back up. Will try to get this merged soon.

felicianotech commented:
Hi @naftulikay,

The failed builds are specifically with the `node` variant images. We recently fixed those in `staging`so maybe that will help.

Can you please do the following?

1. Rebase your branch on the latest version of `staging`. Assuming that this remote is called `origin` on your system, you can do the following: `git rebase -i origin/staging`
2. Use `apt-get` in your PR instead of `apt`.

I predict that these two changes will be all we need to merge. Thank you in advance.

naftulikay commented:
@felicianotech I have rebased on the `staging` branch of `circleci/circleci-images`, so it should be up to date now, and I have replaced `apt` with `apt-get`.

felicianotech commented:
Thank you.

The Rust image is still failing. I'm thinking that this is an issue with our build system and not this PR though. Investigating.
